operators/multiclusterobservability/pkg/servicemonitor/sm_controller.go

```diff
@@ -0,0 +1,155 @@
+		return
+	}
+	isSmControllerRunnning = true
+
+	promClient, err := promclientset.NewForConfig(ctrl.GetConfigOrDie())
+	if err != nil {
+		log.Error(err, "Failed to create prom client")
+		os.Exit(1)
+	}
+	watchlist := cache.NewListWatchFromClient(promClient.MonitoringV1().RESTClient(), "servicemonitors", config.GetDefaultNamespace(),
+		fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		&promv1.ServiceMonitor{},
+		time.Minute*60,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    onAdd(promClient),
+			DeleteFunc: onDelete(promClient),
+			UpdateFunc: onUpdate(promClient),
+		},
+	)
+
+	stop := make(chan struct{})
+	go controller.Run(stop)
+}
+
+func onAdd(promClient promclientset.Interface) func(obj interface{}) {
+	return func(obj interface{}) {
+		sm := obj.(*promv1.ServiceMonitor)
+		if sm.ObjectMeta.OwnerReferences != nil && sm.ObjectMeta.OwnerReferences[0].Kind == "Observatorium" {
+			updateServiceMonitor(promClient, sm)
+		}
+	}
+}
+
+func onDelete(promClient promclientset.Interface) func(obj interface{}) {
+	return func(obj interface{}) {
+		sm := obj.(*promv1.ServiceMonitor)
+		if sm.ObjectMeta.OwnerReferences != nil && sm.ObjectMeta.OwnerReferences[0].Kind == "Observatorium" {
+			err := promClient.MonitoringV1().ServiceMonitors(ocpMonitoringNamespace).Delete(context.TODO(), sm.Name, metav1.DeleteOptions{})
+			if err != nil {
+				log.Error(err, "Failed to delete ServiceMonitor", "namespace", ocpMonitoringNamespace, "name", sm.Name)
+			} else {
+				log.Error(err, "ServiceMonitor Deleted", "namespace", ocpMonitoringNamespace, "name", sm.Name)
+			}
+		}
+	}
+}
+
+func onUpdate(promClient promclientset.Interface) func(newObj interface{}, oldObj interface{}) {
+	return func(newObj interface{}, oldObj interface{}) {
+		newSm := newObj.(*promv1.ServiceMonitor)
+		oldSm := oldObj.(*promv1.ServiceMonitor)
+		if newSm.ObjectMeta.OwnerReferences != nil && newSm.ObjectMeta.OwnerReferences[0].Kind == "Observatorium" &&
+			!reflect.DeepEqual(newSm.Spec, oldSm.Spec) {
+			updateServiceMonitor(promClient, newSm)
+		}
+	}
+}
+
+func updateServiceMonitor(promClient promclientset.Interface, sm *promv1.ServiceMonitor) {
+	found, err := promClient.MonitoringV1().ServiceMonitors(ocpMonitoringNamespace).Get(context.TODO(), sm.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			_, err := promClient.MonitoringV1().ServiceMonitors(ocpMonitoringNamespace).Create(context.TODO(), rewriteLabels(sm), metav1.CreateOptions{})
+			if err != nil {
+				log.Error(err, "Failed to create ServiceMonitor", "namespace", ocpMonitoringNamespace, "name", sm.Name)
+			} else {
+				log.Info("ServiceMonitor Created", "namespace", ocpMonitoringNamespace, "name", sm.Name)
+			}
+		} else {
+			log.Error(err, "Failed to check ServiceMonitor", "namespace", ocpMonitoringNamespace, "name", sm.Name)
+		}
+	}
+	update := rewriteLabels(sm)
+	update.ObjectMeta.ResourceVersion = found.ObjectMeta.ResourceVersion
```

Comment:
We do not need to reset the `ResourceVersion`. It may lead updating conflicted. refer to https://github.com/open-cluster-management/multicluster-observability-operator/pull/721 @morvencao

Reply:
fixed